src/js/common/components/LoginModal.jsx

```diff
@@ -11,17 +11,35 @@ class LoginModal extends React.Component {
   constructor(props) {
     super(props);
     this.handleLogin = this.handleLogin.bind(this);
+
+    // I don't like this, but we need to make sure componentWillReceiveProps
+    //  doesn't call onLogin() when a page with a LoginModal is refreshed and
+    //  a user is logged in.
+    this.state = {
+      loginButtonClicked: false
+    };
   }
 
   componentWillReceiveProps(nextProps) {
     // If the loggedIn status went from false to true, close the modal
     const wasLoggedIn = this.props.user.login.currentlyLoggedIn;
     const isLoggedIn = nextProps.user.login.currentlyLoggedIn;
-    if (!wasLoggedIn && isLoggedIn) {
+    if (!wasLoggedIn && isLoggedIn && this.state.loginButtonClicked) {
+      if (this.props.onLogin) {
+        this.props.onLogin();
+      }
+      this.setState({ loginButtonClicked: false });
       this.props.onClose();
     }
   }
 
+  shouldComponentUpdate(nextProps) {
```

Comment:
This is fine, but it's worth noting that we have recompose, so you could use [onlyUpdateForKeys](https://github.com/acdlite/recompose/blob/master/docs/API.md#onlyupdateforkeys), too.